src/MathCwfs.cc

```
@@ -0,0 +1,1156 @@
+
+            temp += Z[4] * sqrt_6 * 2 / den3;
+            temp += Z[5] * 0;
+
+            temp += Z[6] * sqrt_8 * (1 + e2) * (6 * x_c) / den4;
+            temp += Z[7] * sqrt_8 * 6 * y_c * (1 + e2) / den4;
+
+            temp += Z[8] * sqrt_8 * 6 * x_c / den5;
+            temp += Z[9] * sqrt_8 * (-6) * y_c / den5;
+
+            temp += Z[10] * sqrt_5 * 48 * xy / den6;
+
+            temp += Z[11] * sqrt_10 * 0;
+            temp += Z[12] * sqrt_10 *
+                    ((24 * x2 + 24 * y2 - 6) * (1 + e2 + e4) - 6 * e6) * num7 /
+                    den7;
+
+            temp += Z[13] * sqrt_10 * (-24) * xy / den8;
+            temp += Z[14] * sqrt_10 * 12 * (x2 - y2) / den8;
+
+            temp +=
+                Z[15] * sqrt_12 *
+                (-8 * y_c *
+                 (3 * e6 - 15 * e4 * x2 - 5 * e4 * y2 + 12 * e4 - 60 * e2 * x2 -
+                  20 * e2 * y2 + 12 * e2 - 15 * x2 - 5 * y2 + 3)) *
+                num9 / den9;
+            temp +=
+                Z[16] * sqrt_12 *
+                (-8 * x_c *
+                 (3 * e6 - 5 * e4 * x2 - 15 * e4 * y2 + 12 * e4 - 20 * e2 * x2 -
+                  60 * e2 * y2 + 12 * e2 - 5 * x2 - 15 * y2 + 3)) *
+                num9 / den9;
+
+            temp += Z[17] * sqrt_12 *
+                    (12 * y_c *
+                     (2 * e8 - 5 * e6 * r2 + 2 * e6 - 5 * e4 * r2 + 2 * e4 -
+                      5 * e2 * r2 + 2 * e2 - 5 * r2 + 2)) *
+                    num10 / den10;
+            temp += Z[18] * sqrt_12 *
+                    (-12 * x_c *
+                     (2 * e8 - 5 * e6 * r2 + 2 * e6 - 5 * e4 * r2 + 2 * e4 -
+                      5 * e2 * r2 + 2 * e2 - 5 * r2 + 2)) *
+                    num10 / den10;
+
+            temp += Z[19] * sqrt_12 * 20 * y_c * (y2 - 3 * x2) / den11;
+            temp += Z[20] * sqrt_12 * 20 * x_c * (x2 - 3 * y2) / den11;
+
+            temp += Z[21] * sqrt_7 * 240 * xy * (2 * r2 - 1 - e2) / den12;
+
+            d[ii] = temp;
+        }
+    }
```

Please raise an exception if axis is none of the above.

This is another case where making `axis` an enum class is worth considering.

---

Updated in commit: 6d31edc. Thanks!